Doc/reference/expressions.rst

```diff
@@ -1843,7 +1843,8 @@ object.  The unnamed object behaves like a function object defined with:
 
 See section :ref:`function` for the syntax of parameter lists.  Note that
 functions created with lambda expressions cannot contain statements or
-annotations.
+annotations. See :doc:`standard Python grammar <./grammar>`
+for precise grammar specification of lambda parameters.
```

Comment:
Re-reading, this text may be superfluous: the preceding line already states that "lambda expressions cannot contain ... annotations".

What you could perhaps do is add a special-cased `lambda_parameter_list` definition to the production list here, though it might be a little over the top.

A

Reply:
It would have been good if we could just point to the lambda_parameter definitions in the production list. Right now the docs provide the info that lambda parameters do not accept annotations but if you follow the grammar rules written here you end up with another conclusion.

Not sure what is the best to do here. Copying the lambda_parameter_list from grammars to here seems like a lot of duplication. But at least it's correct.

Reply:
Note that the expressions reference doesn't follow the grammar w.r.t. names of productions. I think if you really want to disambiguate, add the specialised parameter list, but I'm unsure why the admonition in prose that this is banned isn't enough?

A